os/services/lwm2m/lwm2m-engine.c

```diff
@@ -80,10 +80,24 @@
 #define USE_RD_CLIENT 1
 #endif /* LWM2M_ENGINE_CONF_USE_RD_CLIENT */
 
+
+#if LWM2M_Q_MODE_ENABLED
```

Comment:
Any particular reason why not just compile everything, always? I would just get rid of _all_ those conditionals in this file. The linker can garbage collect objects where possible

Reply:
My main goal was to not increase the memory footprint when people wants to use LwM2M but not operate with the Queue Mode at all. Maybe it's not so critical and everything should be compiled... I guess we need to reach an agreement here. @joakimeriksson and @nfi what do you think about it?

Reply:
OK I won't insist, happy with whatever you lot decide

Reply:
I agree that we should try to avoid conditionals in the C-files as much as possible in the long run. Possibly by separating some parts into different files or even starting to accept larger code-base as typical chips are growing and we are kind of giving up on fitting things on the Tmote Sky now. Niclas and I (and Carlos) can try to make a PR with some refactorisation towards avoiding most of the conditionals in the LWM2M code.

Reply:
Related: https://github.com/contiki-ng/contiki-ng/issues/248

Reply:
If we consider that the memory footprint is not critical and all the Queue Mode code can just be compiled always with LwM2M, then there is no point on keeping the preprocesor conditionals. I think is a good solution to keep the code as it is for now, and make another PR to remove them.

Also, as @joakimeriksson says, we can move all the Q-Mode functions defined in `lwm2m-engine.[hc]` and `lwm2m-rd-client.[hc]` to other files if you think that would improve the code, and just keep one big conditional in those files, as in `lwm2m-qmode-object` or `lwm2m-notification-queue`. I think that is fast and I can even do it for this PR, or maybe we can just take care of everything in a future PR... what do you think?